eclair-core/src/main/scala/fr/acinq/eclair/channel/ChannelData.scala

```diff
@@ -417,6 +417,11 @@ sealed trait LocalFundingStatus {
   def signedTx_opt: Option[Transaction]
   /** We store local signatures for the purpose of retransmitting if the funding/splicing flow is interrupted. */
   def localSigs_opt: Option[TxSignatures]
+  /**
+   * Commitment index of the first remote commitment we signed that spends this funding transaction.
+   * Once the funding transaction confirms, our peer won't be able to publish revoked commitments with lower indices.
+   */
+  def firstCommitIndex_opt: Option[Long]
```

Comment:
I think this could be in the main `Commitment` class, it feels a bit out of place here.  We could also maybe make it non-optional, with a proper comment?

At first I thought we could even do without it entirely, by looking at pruned commitments, but it's actually orthogonal.

Reply:
Done in https://github.com/ACINQ/eclair/pull/2705/commits/bc075481bb67531e3aa2596c3a844c4ab38bd192, I initially put it in the `FundingStatus` to minimize the codec changes, but it's actually just a "dumb" copy-paste so it's not too bad. This way we also add a version to the commitments codec to make such changes easier in the future.